src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/Metadata/IModelBindingMessageProvider.cs

```diff
@@ -25,6 +25,13 @@ public interface IModelBindingMessageProvider
         Func<string> MissingKeyOrValueAccessor { get; }
 
         /// <summary>
+        /// Error message the model binding system adds when no value is provided for the request body,
+        /// but a value is required.
+        /// </summary>
+        /// <value>Default <see cref="string"/> is "A value for the request body was not provided.".</value>
+        Func<string> MissingRequestBodyRequiredValueAccessor { get; }
```

Comment:
At the moment, `IModelBindingMessageProvider` is in `...Mvc.Abstractions`, whereas the implementation `ModelBindingMessageProvider` is in `...Mvc.Core`.

Other code in `...Abstractions` references `IModelBindingMessageProvider`. So to remove that interface, we'd need to move `ModelBindingMessageProvider` across into `...Abstractions`, along with its `.resx` file.

Of course we can do this, but I'd prefer to regard it as an independent change, because I don't personally know how we manage localisation (e.g., do we have some build process that's configured to do something with a known set of `.resx` files, to ensure all the different language translations get included in the final builds? how would they need to be updated if we moved a `.resx` file to a different project?).

So I'm regarding this as "won't fix" (as a CR feedback item on issue, that is) unless anyone feels strongly enough and can describe what would have to happen to keep localisation showing up properly in builds after moving the `.resx` file. Hope that's OK!